homeassistant/components/plant.py

```diff
@@ -244,4 +319,21 @@ def state_attributes(self):
         for reading in self._sensormap.values():
             attrib[reading] = getattr(self, '_{}'.format(reading))
 
+        if self._max_brightness_over_days is not None:
+            attrib['max brightness over {} days'.format(
+                self._conf_check_days)] = self._max_brightness_over_days
+
         return attrib
+
+    @asyncio.coroutine
+    def async_added_to_hass(self):
+        """Create a group with all sensors.
+
+        This must be run after the component was created to that we get the
+        valid entity_id.
+        """
+        if self._group_name is not None:
+            members = [self.entity_id]
+            members.extend(list(self._config[CONF_SENSORS].values()))
+            self.hass.components.group.set_group(self._group_name,
+                                                 entity_ids=members)
```

Comment:
You can that do more simple with: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/group/__init__.py#L169

Reply:
@pvizeli 

1) From looking at the code you linked, I'm not sure how you want to simplify this. Please be more specific.
2) This is already the 4th version I've implemented to create these groups. This is quite frustrating for me to keep changing such a simple feature without benefit for the users. So please be more specific in your code reviews and describe more clearly what you would like to have.
3) Why are there so many different ways to create groups in HASS? And if only one of these ways is correct, then please provide some documentation so that other developers use the right ones, especially update  "demo.py".

Reply:
@pvizeli 
Since I did not hear from you: I just tried to use "add=members" instead of "entity_ids"=members but that does not provide the results I wanted to have: add seems to add the elements to the group in a random order, but I want the plant component to be the first element in the list.

So I'm still not sure what you want...